internal/types/document_validation.go

```diff
@@ -115,7 +120,7 @@ func (d *Document) ValidateData() error {
 				return err
 			}
 		case *Array:
-			if key == "_id" {
+			if isTopLevel && key == "_id" {
```

Comment:
@slavabobik codecov still marks some lines uncovered, it's because we don't have tests where we try to insert documents with top-level `_id` set as array or regex. A good candidate for such tests could be `TestInsertCommandErrors`, we can add examples where top-level `_id` field is filled with those invalid types and we know which error we expect. Personally, I would add such tests, but in this PR I leave it up to you to do it (if you have more time and wish, please feel free to do it, otherwise we can do it separately). 

What else we don't have is a test case where top-level `_id` is set as a document (a valid case, but probably rate in real life), but it's rather a separate story :)

Reply:
@rumyantseva I can add more tests in separate PR :) Do I need a separate task or just create a PR with tests?

Reply:
Let‘s have a separate PR only. A separate task is not needed. Thank you!